### PR TITLE
SG - Added BCrypt to User Passwords

### PR DESCRIPTION
**Pull request by StevenGarberg**

### CODE REVIEW

File: Controllers/UsersController.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public async Task<User> Create(UserCreateRequest request)
             {
                 Id = Guid.NewGuid().ToString(),
                 Username = request.Username,
-                Password = request.Password,
+                Password = BCrypt.Net.BCrypt.HashPassword(request.Password),
```
Comment (StevenGarberg) [inline Controllers/UsersController.cs:42]: @christopherplummer This is the main change.

File: Controllers/UsersController.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public async Task<User> Update(Guid id, UserUpdateRequest request)
 
             if (!string.IsNullOrEmpty(request.Password))
             {
-                user.Password = request.Password;
+                user.Password = BCrypt.Net.BCrypt.HashPassword(request.Password);
```
Comment (StevenGarberg) [inline Controllers/UsersController.cs:62]: @christopherplummer Same as last comment.

File: FriendlyApi.Service.csproj [hunk 3]
```diff
@@ -5,6 +5,8 @@
     </PropertyGroup>
 
     <ItemGroup>
+        <PackageReference Include="BCrypt" Version="1.0.0" />
+        <PackageReference Include="BCrypt.Net-Next" Version="4.0.2" />
```
Comment (StevenGarberg) [inline FriendlyApi.Service.csproj:9]: @christopherplummer Gotta import mah libraries.

**Comment by StevenGarberg** [comment]
@christopherplummer A lot of this PR is just noise with me rearranging things and improving the `.gitignore`. Please take a look at this and approve if you're good with it. Thanks!

**Comment by StevenGarberg** [comment]
@christopherplummer Also, this is just the hashing for incoming passwords for now. We can add an authentication endpoint to compare an incoming value to the existing hash later.

**Comment by christopherplummer** [comment]
GOOD LORD THIS IS EXCELLENT WORK SON, WHY DOESN'T SOMEONE HIRE YOU?!?!?!

**Review by christopherplummer — APPROVED** [review]
Yay for security!

**Pull request merged**